Framework/NexusGeometry/src/NexusGeometrySave.cpp

```diff
@@ -277,12 +264,21 @@ inline void writeXYZPixeloffset(H5::Group &grp,
  * @param compInfo : componentInfo object.
  */
 inline void writeDetectorNumber(H5::Group &grp,
-                                const Geometry::DetectorInfo &detInfo) {
+                                const Geometry::ComponentInfo &compInfo,
+                                const size_t &idx,
+                                const std::vector<int> &detectorIDs) {
 
   H5::DataSet detectorNumber;
 
-  std::vector<int> detIDs = detInfo.detectorIDs();
-  const hsize_t dimSize = (hsize_t)detIDs.size();
+  std::vector<int> bankDetIDs;
+  std::vector<size_t> bankDetectors = compInfo.detectorsInSubtree(idx);
+  bankDetIDs.reserve(bankDetectors.size());
+
+  for (size_t &index : bankDetectors) {
+    bankDetIDs.push_back(detectorIDs[index]);
+  }
+
+  const hsize_t dimSize = (hsize_t)bankDetIDs.size();
```

Comment:
Solution over all looks good, but do not use c-style casts. `(hsize_t)`. Good chapter about this in dewhurst book. change to `const auto nDetectorsInBank = static_cast<hsize_t>(bankDetIDs.size())`.